### PR TITLE
Module base_module_doc_rst migrated from version 6.1

### PR DESCRIPTION
**Pull request by sysadminmatmoz**
I use it as a tool for the relationship graph mostly, but it could be handy for someone else too.

### CODE REVIEW

**Comment by sysadminmatmoz** [comment]
I cleaned what I could (following travis), for the rest I'll need some help.

File: base_module_doc_rst/wizard/tech_rst_guide.py [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+            logger = netsvc.Logger()
+            msg = "Object %s not found" % (obj)
+            logger.notifyChannel("base_module_doc_rst", netsvc.LOG_ERROR, msg)
+            return ""
+
+    _defaults = {
+        'rst_file': _generate,
+    }
+
+WizardTechGuideRst()
```
Comment (moylop260) [inline base_module_doc_rst/wizard/tech_rst_guide.py]: Now you don't need invoke to classes

File: base_module_doc_rst/README.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License AGPL-3
+
+This module generates the Technical Guides of selected modules in Restructured Text format (RST).
```
Comment (pedrobaeza) [inline base_module_doc_rst/README.rst:6]: Put upper line also for highlighting the title. Remove final point

File: base_module_doc_rst/base_module_doc_rst.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        os.popen(
+            'dot -Tpng' + ' ' + path_dotfile + ' ' + '-o' + ' ' + path_png
+        )
+        fp = file(path_png, "r")
+        x = fp.read()
+        fp.close()
+        os.popen('rm ' + path_dotfile + ' ' + path_png)
+        return {'module_file': base64.encodestring(x)}
+
+Module()
```
Comment (pedrobaeza) [inline base_module_doc_rst/base_module_doc_rst.py]: This isn't needed anymore

File: base_module_doc_rst/wizard/generate_relation_graph.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import osv
+
+
+class CreateGraph(osv.osv_memory):
```
Comment (pedrobaeza) [inline base_module_doc_rst/wizard/generate_relation_graph.py]: Put orm.TransientModel

File: base_module_doc_rst/wizard/generate_relation_graph.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                mod_obj.write(
+                    cr,
+                    uid,
+                    [module.id],
+                    {'file_graph': module_data['module_file']},
+                    context=context
+                )
+        return {'type': 'ir.actions.act_window_close'}
+
+CreateGraph()
```
Comment (pedrobaeza) [inline base_module_doc_rst/wizard/generate_relation_graph.py]: Remove this

File: base_module_doc_rst/wizard/tech_rst_guide.py [hunk 6]
```diff
@@ -0,0 +1,525 @@
...
+        s += self._write_reports()
+        s += self._write_menus()
+        s += self._write_views()
+        s += self._write_objects()
+        if module_name:
+            s += self._write_relationship_graph(module_name)
+        return s
+
+
+class WizardTechGuideRst(osv.osv_memory):
```
Comment (pedrobaeza) [inline base_module_doc_rst/wizard/tech_rst_guide.py]: orm.TransientModel

**Comment by pedrobaeza** [comment]
Check pylint error:

```
base_module_doc_rst/wizard/tech_rst_guide.py:55: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\*'. String constant might be missing an r prefix.
```

File: base_module_doc_rst/wizard/generate_relation_graph.py (hunk 4 above)
Comment (sebalix) [inline base_module_doc_rst/wizard/generate_relation_graph.py]: In v8, is not recommended to use this:

``` python
from openerp import models

class CreateGraph(models.TransientModel):
```

?

Comment (pedrobaeza) [inline base_module_doc_rst/wizard/generate_relation_graph.py]: That's new API. Not everyone wants to migrate the code to it, and it's not required.

File: base_module_doc_rst/wizard/generate_relation_graph.py (hunk 5 above)
Comment (sysadminmatmoz) [inline base_module_doc_rst/wizard/generate_relation_graph.py]: Did you mean the whole section from line 33?

Comment (pedrobaeza) [inline base_module_doc_rst/wizard/generate_relation_graph.py]: No, only this line that it's useless

**Comment by sysadminmatmoz** [comment]
That should do it.

**Comment by sysadminmatmoz** [comment]
Ok, failed builds are not relative to this commit:
************\* Module auth_from_http_remote_user.**openerp**
auth_from_http_remote_user/**openerp**.py:21: [W0109(duplicate-key), ] Duplicate key 'license' in dictionary
************\* Module disable_openerp_online.**openerp**
disable_openerp_online/**openerp**.py:27: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\P'. String constant might be missing an r prefix.

**Comment by pedrobaeza** [comment]
You still have an error:

```
./base_module_doc_rst/base_module_doc_rst.py:85:18: F402 import 'fields' from line 24 shadowed by loop variable
```

**Comment by sysadminmatmoz** [comment]
Done. You made me set up flake and pylint on my local machine to find out what was the problem! ... And they say that an old dog can't learn new tricks! :smile:

**Comment by dhp-denero** [comment]
he he he.

Thanks,

Dhaval Patel
Team Lead Engineer
Denero Team.
www.deneroteam.com
Skype : deneroteam
Mobile : +91-9327694716

**Comment by moylop260** [comment]
@sysadminmatmoz 
You can use [git-hook](https://github.com/OCA/maintainer-quality-tools/pull/242) to automate in local lint checks.

File: base_module_doc_rst/wizard/generate_relation_graph.py (hunk 4 above)
Comment (moylop260) [inline base_module_doc_rst/wizard/generate_relation_graph.py]: @pedrobaeza and @sebalix 
This check is enabled here:
https://github.com/OCA/maintainer-quality-tools/pull/243
Your feedback is welcome

**Comment by pedrobaeza** [comment]
OK, current errors come from another modules. :+1:

**Comment by sysadminmatmoz** [comment]
So why not correcting the other modules since I'm here. :smile:

**Comment by pedrobaeza** [comment]
Thanks for your efforts! It seems that something strange has come in your work, because there are some commits that are mixed that doesn't belong to your PR.

Trying your PR in runbot, I can't get the relationship graph. Sometimes it doesn't say anything, but no image is presented, and sometimes I get this error:

```
IOError: [Errno 2] No such file or directory: u'/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3115452-253-45af56/openerp/addons/base_module_doc_rst/module.png'
```

And the RST generated is in an unreadable format. It's not plain RST?

**Comment by sysadminmatmoz** [comment]
The image gets presented only for the modules that are really currently installed (not just present as installable) in the same database where you run the creation button. The RST is the sphinx RST (as defined in description) - **the output file is a gzip archive** containing index.rst, module.rst and relations graph.

What commits not belonging to the PR do you mean? 2 files are just corrections to modules that Travis complained about, 2 are relative to a merge, since the repository was updated since I made the original PR.

Regarding the issue: could it be a rebase problem? It works quite fine on a real instance DB.

relation_graph

and sphinx output after make html:
sphinx1

**Comment by pedrobaeza** [comment]
Thanks for the explanations. Can you perform then these operations:
- Hide "Relationship Graph" page if module not installed.
- Raise an exception when clicking on "Generate Relationship..." if the module is not installed.

**Comment by sysadminmatmoz** [comment]
That's one of the reasons I became a member of the community. There are some things I'm good at and some things where I'm not, but I'm sure that someone else in the community is. Professionally I'm a problem solver, I'm good at making things work; but to polish them to the final product I need a specialist, thus someone else :smile: .

I'm sure I'd be able to do what you ask eventually (spending hours of study and experiments as with everything else regarding coding, since it's just a hobby for me and not what I usually really do), but there's sure someone in the community that already knows exactly what to do spending minutes where I would spend hours. So, to make it short: if those operations are a must to get things moving, I'll do it (in a month or so, when I'll have some time to spend on it), but I really hope that someone better than me gets his hands on it way before that. :smile:

**Comment by pedrobaeza** [comment]
Well, I was only suggesting that improvements, but I'm not very interested for now in the module itself, so if you can't work on them, at least annotate them on "Know issues" section, and see if any other can improve them.

**Comment by sysadminmatmoz** [comment]
Moving the PR in https://github.com/OCA/server-tools/pull/257 and closing, since I messed up my repo.

**Pull request closed without merging**